### PR TITLE
Add initial cucumber tests to alarm service with docker containers

### PR DESCRIPTION
**Pull request by mfrazier63**
## Description
This PR has the initial work to add in cucumber integration tests for the alarm service.

Utilizes docker images in the docker-maven-plugin to spin up the microservice with all it's dependencies (kafka, postgress, etc) so that we can interact with the alarm service through kafka and REST and examine the results.

For now has just one simple test case that sends and event to kafka, then makes sure that an alarm was created and persisted.

## Jira link(s)
- https://issues.opennms.org/browse/HS-xx

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: Tiltfile [hunk 1]
```diff
@@ -126,7 +126,7 @@ jib_project(
     'opennms/horizon-stream-alarm',
     'alarm',
     'opennms-alarm',
-    port_forwards=['29080:9090', '29050:5005', '29065:6565',  '29000:8080'],
+    port_forwards=['31080:9090', '31050:5005', '31065:6565',  '31000:8080'],
```
Comment (thomasbigger584) [inline Tiltfile:129]: think we've to change this to 32xxx actually because it looks like the new proxy service took 31xxx

Comment (mfrazier63) [reply Tiltfile]: yeah i just picked that up when back merging develop

**Review by geraldhumphries — APPROVED** [review]

File: alarm/src/main/java/org/opennms/horizon/alarmservice/rest/AlarmRestServiceImpl.java [hunk 2]
```diff
@@ -98,32 +96,60 @@ public Response getAlarms(@Context final SecurityContext securityContext, @Conte
...
-            return Response.status(Status.OK).entity(alarmsCollection).build();
+            return ResponseEntity.ok(alarmsCollection);
     }
     
-    @PostMapping(path="{id}/clear", produces = MimeTypeUtils.APPLICATION_JSON_VALUE)
-    public String clearAlarm(@PathVariable Long id) {
+    @PostMapping(path="/clear/{id}", produces = MimeTypeUtils.APPLICATION_JSON_VALUE)
+    public ResponseEntity<AlarmDTO> clearAlarm(@PathVariable Long id) {
+
+        return ResponseEntity.ok(alarmService.clearAlarm(id, new Date()));
```
Comment (artnaseef) [inline alarm/src/main/java/org/opennms/horizon/alarmservice/rest/AlarmRestServiceImpl.java:105]: Nit here.  I personally prefer to perform the operation on one line, then wrap the result on another, even though Intellij complains the local variable can be removed.  I find it easier to read and debug.

Just felt like sharing.  No need to make any change.

File: alarm/src/main/resources/org/opennms/horizon/dockerit/alarm.feature [hunk 3]
```diff
@@ -0,0 +1,43 @@
+Feature: Alarm Service Basic Functionality
```
Comment (artnaseef) [inline alarm/src/main/resources/org/opennms/horizon/dockerit/alarm.feature:1]: Was this meant to be left commented-out?

Ahh, see the other one in src/tes/resources.  This should just be removed, right?

**Pull request merged**